### DISCUSSION TITLE
Use Write function but return "cannot estimate gas"

### DISCUSSION THREAD

**Discussion by WRyv** [wevm/wagmi#617] [Q&A]
Hello 

i'm using 
const { data, isError, isLoading, write } = useContractWrite(
    {
      addressOrName: "0x9F_______",
      contractInterface: AbiDistributor,
    },
    "claimDividend"
  );

for claim token in a smart contract

i have result: next-dev.js?3515:25 Error: cannot estimate gas; transaction may fail or may require manual gas limit [ See: https://links.ethers.org/v5-errors-UNPREDICTABLE_GAS_LIMIT ] (reason="execution reverted" 

If i'm using a other function in the same smart contrast is working

How can i to do for know the problem?

Thank you !

**Comment by Jiliac** [comment]
https://www.reddit.com/r/ethdev/comments/sybe22/can_anyone_explain_this_error_error_cannot/

This thread says the simplest way to solve this problem is by setting a gas limit.

`overrides: { gasLimit: 1e7 }`

I was having the same problem and I think it works. (I'm hitting another bug ATM.)

**Reply by mmatteo23** [reply]
Hi! But with this override wagmi can't help you to check if an error occurs I think. I tried this solution because I need to prevent wagmi from throwing unhandled errors in usePrepareContractWrite() and useContractWrite(). So with that override wagmi returns no error and tries to process the transaction anyway (which obviously fails)

**Reply by halalanft** [reply]
@mmatteo23 do you have any other solution?

**Reply by mmatteo23** [reply]
It's been a while since I've used wagmi and I used it only for a proof-of-concept creation. Anyway, when you have this type of error, it means that there is a configuration problem with the contract call (for example wrong contract, unauthorized address, etc...).
In my case, I wasn't able to prevent the throwing of the error but I decided to accept it since many dApps don't block Metamask errors from being thrown.
If you increase gas limit to that level you can no longer control whether the call will fail (for example usePrepareContractWrite will always return no errors) so the user will have the error at a later time directly on the wallet without well understanding the reason for the failure.

**Reply by halalanft** [reply]
@mmatteo23 what do you use lately, if I may know?

**Reply by 0xCardiE** [reply]
I am having the same problem with this code, until I put 
   gasLimit: 1e7,
   function was failing, also it worked perfectly if I didnt use overrides and sent some ether. Not sure what is wrong here, but this gasLimit setting does help.

```
  const { config, error } = usePrepareContractWrite({
    ...contractConfig,
    functionName: "mint",
    args: [mintQuantity],
    overrides: {
      gasLimit: 1e7,
      value: ethers.utils.parseUnits(payment.toString()),
    },
  });
```

**Reply by halalanft** [reply]
The gas limit is not solved the problem completely. What I did is to use `usePrepareContractWrite`  and make it sure it is disabled until we have the exact parameter like so:
```
    const [baseURI, setBaseURI] = useState("");
    const {
        config: configBaseURI,
    } = usePrepareContractWrite({
        address: contractAddres.ProudCamel,
        abi: proudABI.abi,
        functionName: 'setBaseURI',
        args: [baseURI],
        enabled: props.isOwner && Boolean(baseURI),
        onError(error) {
            console.log('Error BaseURI', error)
        },
    })

    const {
        write: writeBaseURI
    } = useContractWrite(configBaseURI)
```

**Reply by 0xCardiE** [reply]
I have all the parameters from start in usePrepareContractWrite but still, it can't determine gas on Goerli by itself so I need to put some overrides, otherwise, I get a not really helpful message that the functionName I use to trigger write is not a function.
